pkg/api/router.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2021 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"context"
+	"path"
+
+	v1 "github.com/GoogleCloudPlatform/testgrid/pkg/api/v1"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/gorilla/mux"
+)
+
+// RouterOptions are the options needed to GetRouter
+type RouterOptions struct {
+	GcsCredentials string
+	Hostname       string
+	HomeBucket     string
+}
+
+const v1Infix = "/api/v1"
```

Comment:
Please move inside GetRouter where it is used (aka keep it scoped to `GetRouter()`)